project/views.py

```diff
@@ -1,14 +1,117 @@
 
-from django.http        import JsonResponse
-from django.views       import View
+from django.db        import transaction
+from django.views     import View
+from django.conf      import settings
+from django.http      import JsonResponse
 
-from user.utils         import user_decorator
-from .models            import Category, Project, Gift, Community, Story, Like
-from user.models        import User
-from message.models     import Message
+from my_settings      import (
+    SECRET_KEY, ALGORITHM,
+    AWS_ID    , AWS_KEY
+)
+
+from user.models      import User
+from user.utils       import login_decorator, user_decorator
+from user.models      import User
+from .models          import  (
+    Category, Project, Like,
+    Gift, Story, Community
+)
+     
+class RegisterView(View):   
+    @login_decorator
+    def post(self, request):
+        data    = json.loads(request.body)
+        user_id = request.user.id
+
+        try:
+            with transaction.atomic():
+                name          = data['name']
+                thumbnail_url = data['thumbnail_url']
+                summary       = data['summary']
+                category      = data['category']
+                story         = data['story']
+                goal_amount   = data['goal_amount']
+                total_amount  = data['total_amount']
+                opening_date  = data['opening_date']
+                closing_date  = data['closing_date']
+                project_uri   = data['project_uri']
+                gifts         = data['gifts']
+
+                category_id = Category.objects.get(name=category).id
+
+                project = Project.objects.create(
+                    category_id   = category_id,
+                    user_id       = user_id,
+                    name          = name,
+                    opening_date  = datetime.strptime(opening_date,'%Y-%m-%d'),
+                    closing_date  = datetime.strptime(closing_date,'%Y-%m-%d'),
+                    thumbnail_url = thumbnail_url,
+                    goal_amount   = goal_amount,
+                    total_amount  = total_amount,
+                    summary       = summary,
+                    project_uri   = project_uri
+                )
+
+                Story.objects.create(
+                    content    = story,
+                    project_id = project.id
+                    )
+
+                for gift in gifts: 
+                    gift_name     = gift['gift_name']
+                    gift_price    = gift['gift_price']
+                    gift_stock    = gift['gift_stock']
+                    quantity_sold = gift['quantity_sold']
+
+                    Gift.objects.create(
+                        name          = gift_name,
+                        price         = gift_price,
+                        stock         = gift_stock,
+                        project_id    = project_id,
+                        quantity_sold = quantity_sold
+                    )
```

Comment:
```suggestion

                for gift in gifts: 
                    Gift.objects.create(
                        name          = gift['gift_name'],
                        price         = gift['gift_price'],
                        stock         = gift['gift_stock'],
                        quantity_sold = gift['quantity_sold'],
                        project_id    = project_id
                        
                    )
```
반복해서 사용하지 않고, 가독성도 해치지 않는 것 같으니 위와 같이 수정하는 것이 더 가독성이 좋아 보입니다.